main.go

```diff
@@ -83,15 +85,24 @@ func main() {
 		log.Error(err, "unable to complete/verify oidc setup for IRSA")
 	}
 
-	if err = (&controllers.IamroleReconciler{
+	controller := &controllers.IamroleReconciler{
 		Client:    mgr.GetClient(),
 		IAMClient: iamClient,
 		Recorder:  k8s.NewK8sClientDoOrDie().SetUpEventHandler(context.Background()),
-	}).SetupWithManager(mgr); err != nil {
+	}
+
+	if err = controller.SetupWithManager(mgr); err != nil {
 		log.Error(err, "unable to create controller", "controller", "Iamrole")
 		os.Exit(1)
 	}
 
+	if err = mgr.Add(manager.RunnableFunc(func(<-chan struct{}) error {
+		return controller.StartControllerReconcileCronJob(context.Background())
```

Comment:
So a few things we may need to:
- update client-go, controller-runtime and api-machinery dependencies.
- runnable func here should block, actually the context should be passed along from RunnableFunc and use that instead of `context.Background`

According to latest [docs](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/manager#Runnable), 
> Runnable allows a component to be started. It's very important that Start blocks until it's done running. 

The issue is you have older dependencies and the args are not using context but they should, we need to upgrade.

Reply:
You may use a go routine or pool in the `StartControllerReconcileCronJob` but by being runnable you accept the same context as manager.

Reply:
> So a few things we may need to:
> 
> * update client-go, controller-runtime and api-machinery dependencies.
> * runnable func here should block, actually the context should be passed along from RunnableFunc and use that instead of `context.Background`
> 
> According to latest [docs](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/manager#Runnable),
> 
> > Runnable allows a component to be started. It's very important that Start blocks until it's done running.
> > Start blocks until the context is closed or an error occurs.
> 
> The issue is you have older dependencies and the args are not using context but they should, we need to upgrade.

Upgraded the dependencies and use newer version of `conroller-runtime` and `client-go`



> You may use a go routine or pool in the `StartControllerReconcileCronJob` but by being runnable you accept the same context as manager.

Updated dependencies, and use context passed from caller.